de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/concept/types/structural/level/HierarchyLevelCheckerTest.java

```diff
@@ -0,0 +1,362 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.concept.types.structural.level;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.concept.types.structural.IBeanStructuralElementInstance;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.extension.tests.model.AConceptTestCase;
+import de.dlr.sc.virsat.model.extension.tests.model.TestStructuralElement;
+
+/**
+ * Test class for {@link HierarchyLevelChecker}
+ */
+public class HierarchyLevelCheckerTest extends AConceptTestCase {
+
+	private Concept concept;
+
+	private IHierarchyLevel a;
+	private IHierarchyLevel b;
+	private IHierarchyLevel c;
+	private IHierarchyLevel an;
+	private IHierarchyLevel bn;
+	private IHierarchyLevel cn;
+	private IHierarchyLevel ao;
+	private IHierarchyLevel bo;
+	private IHierarchyLevel co;
+	private IHierarchyLevel ano;
+
+	@Before
+	public void setup() {
+		concept = loadConceptFromPlugin();
+
+		a = createNameMatchingLevel("a", false, false);
+		b = createNameMatchingLevel("b", false, false);
+		c = createNameMatchingLevel("c", false, false);
+		an = createNameMatchingLevel("a", true, false);
+		bn = createNameMatchingLevel("b", true, false);
+		cn = createNameMatchingLevel("c", true, false);
+		ao = createNameMatchingLevel("a", false, true);
+		bo = createNameMatchingLevel("b", false, true);
+		co = createNameMatchingLevel("c", false, true);
+		ano = createNameMatchingLevel("a", true, true);
+	}
+
+	@Test
+	public void testSingleBean() {
+		HierarchyLevelChecker checker = getChecker(a);
+		IBeanStructuralElementInstance bean = createBean("a");
+
+		assertEquals(expected(a), checker.getApplicableLevels(bean));
+		assertTrue(checker.validateApplicableLevel(bean));
+	}
+
+	@Test
+	public void testSingleBean2() {
+		HierarchyLevelChecker checker = getChecker(a);
+		IBeanStructuralElementInstance bean = createBean("x");
+
+		assertEquals(expected(a), checker.getApplicableLevels(bean));
+		assertTrue(checker.validateApplicableLevel(bean));

```

Comment:
We use the assert statements with mesages, to give a rational about the expected result.

Reply:
I disagree with this (using messages), let's discuss it on the developer meeting.

I agree though that the line is not very clear, but the proper way to fix it is to improve the name `validateApplicableLevel` so that it's clear what it does.